formbricks/formbricks
# Fix/close survey after x responses

## What does this PR do?

This PR fixes the issue where a user was able to enter the limit of close survey after x responses less than or equal to the current number of responses.

Technical Details:
Have updated the useSurvey hook to take in a parameter for analytics where based on that we can query the analytics field from the survey when needed. Have used this analytics to check for the number of responses. 


https://github.com/formbricks/formbricks/assets/47187878/a3c84d9e-5eb0-4286-a325-3baf4376c552



Fixes #561 

## Type of change




## How should this be tested?



- Create a survey
- Add some responses to it
- Now edit this survey 
- Go to setting under the response options
- Click on Close survey on response limit
- See that you are not able to enter a number lesser than your current number of responses.

## Checklist
